DRAFT: Multi-Party Payout.md

```diff
@@ -0,0 +1,204 @@
+
+This DIP defines:
+
+  1. the sharedReward field
+  2. the shareHolderReward field
+  3. corresponding update to the ProRegTx
+  4. corresponding update to the ProUpRegTx
+
+All other transactions, transaction fields, the mechanisms for adding or removing masternodes and defining MN quorum and valid subset remain unchanged.
+
+## Registering a Masternode (ProRegTx)
+
+To join the masternode list, masternode owners must submit a special transaction (DIP2) to the network. DIP 3 defines the ProRegTx and the ProRegUpTx message, parts, and payload: `<magicString><payoutStr>|<operatorReward>|<ownerKeyAddress>|<votingKeyAddress>|<payloadHash>`
+
+We propose adding an additional part (`<sharedReward>`)to produce a message that includes:
+
+`<magicString><payoutStr>|<operatorReward>|<ownerKeyAddress>|<votingKeyAddress>|<sharedReward>|<payloadHash>`
+
+The individual parts of the message are:
+
+1. `<magicString>`: (DIP3) A fixed string which is always "DarkCoin Signed Message:" with a newline character appended.
+Please note that existing tools (e.g. the RPC command `signmessage`) usually add the magicString internally, making it
+unnecessary to manually add it.
+2. `<payoutStr>`: (DIP3) The Dash address corresponding to the scriptPayout field of the ProRegTx. In case scriptPayout is not a
+P2PK/P2PKH/P2SH script, `<payoutStr>` must be set to the hex representation of the full script.
+3. `<operatorReward>`: (DIP3) The operatorReward field of the ProRegTx.
+4. `<ownerKeyAddress>`: (DIP3) The Dash address corresponding to the keyIdOwner field of the ProRegTx.
+5. `<votingKeyAddress>`: (DIP3) The Dash address corresponding to the keyIdVoting field of the ProRegTx.
+6. `<sharedReward>`: (NEW) the sharedReward field of the ProRegTx.
+7. `<payloadHash>`: (DIP3) The SHA256 hash of the ProRegTx payload with the payloadSig being empty.
+
+The ProRegTx discretely specifies 2 rewards, the operatorReward and the sharedReward. The combined calculation of these two rewards then defines the owner's reward:
+
+* The percentage of the masternode reward paid to the operator is calculated by dividing the operatorReward field by 100.
+* The total percentage of the masternode reward to be paid to shareHolders is calculated by dividing the sharedReward field by 100.
+* The remainder of the masternode reward is paid fully to the owner address.
+* The percentage of the sharedReward paid to each shareHolder is the defined in the shareHolders field.
+
+The ProRegTx also specifies a new field, shareHolders, that contains the addresses and percentage of the sharedReward each address is paid. This field is an array of key:value pairs. The key is Dash addresses that will receive rewards. The value is an integer between 1 and 100 (inclusive). This value is the percentage of the sharedReward the address will receive. To mitigate potential block size issues: this field is limited to a length of 100.

```

Comment:
`1 and 100` shouldn't really deviate from the norm of `1-10000` like operator reward

Reply:
Not sure 100 is a good max size.. Should probably be lower than that really

Reply:
how low should the limit be? I don't know block limits or where to find that documentation.

Reply:
I think 32 would be reasonable enough, this would only add about 1kb to a block, while allowing a lot of variability